metacat-connector-druid/src/main/java/com.netflix.metacat.connector.druid/client/DruidHttpClient.java

```diff
@@ -0,0 +1,102 @@
+/*
+ *  Copyright 2017 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+
+package com.netflix.metacat.connector.druid.client;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.netflix.metacat.common.json.MetacatJsonLocator;
+import com.netflix.metacat.common.server.connectors.ConnectorContext;
+import com.netflix.metacat.connector.druid.DruidConfigConstants;
+import com.netflix.metacat.connector.druid.IMetacatDruidClient;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONArray;
+import org.springframework.web.client.RestTemplate;
+
+import javax.annotation.Nullable;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * DruidHttpClient.
+ *
+ * @author zhenl
+ * @since 1.1.0
+ */
+@Slf4j
+public class DruidHttpClient implements IMetacatDruidClient {
+    private String druidURI;
+    private final RestTemplate restTemplate;
+    private final MetacatJsonLocator jsonLocator = new MetacatJsonLocator();
+
+    /**
+     * Constructor.
+     *
+     * @param connectorContext connector context
+     * @param restTemplate     rest template
+     * @throws UnknownHostException exception for unknownhost
+     */
+    public DruidHttpClient(final ConnectorContext connectorContext,
+                           final RestTemplate restTemplate) throws UnknownHostException {
+        this.restTemplate = restTemplate;
+        final String host = connectorContext.getConfiguration().get(DruidConfigConstants.DRUID_SERVER);
+        final String port = connectorContext.getConfiguration().get(DruidConfigConstants.DRUID_PORT);
+
+        if (host != null && port != null) {
```

Comment:
So what happens when one of these is indeed null? You basically don't have an endpoint and then all the methods below blow up? It seems to me if you expect these to not be null you should either provide default values in the get above or throw an exception here that keeps the class from being constructed in a bad state